internal/controller/operatorconfigmap_controller.go

```diff
@@ -398,6 +405,19 @@ func (c *OperatorConfigMapReconciler) reconcileDelegatedCSI() error {
 		return fmt.Errorf("unable to find the updated cluster version")
 	}
 
+	var cniNetworkAnnotationValues []string
+	if len(storageClients.Items) != 0 {
+		for i := range storageClients.Items {
+			annotationValue := storageClients.Items[i].GetAnnotations()[cniNetworksAnnotationKey]
+			if annotationValue != "" {
+				cniNetworkAnnotationValues = append(cniNetworkAnnotationValues, annotationValue)
+			}
+		}
+		if len(cniNetworkAnnotationValues) != 1 {
+			return fmt.Errorf("only one client with CNI network annotation value is supported, got %d", len(cniNetworkAnnotationValues))
+		}
+	}
```

Comment:
- There is no need to check agains 0 items, the range operator will automatically covers it
- There is no need to have a list of annotation of we are to fail after we find the first one
```suggestion
	cniNetworksAnnotationValue : = ""
	for i := range storageClients.Items {
		if value := storageClients.Items[i].GetAnnotations()[cniNetworksAnnotationKey]; value != "" {
			if cniNetworksAnnotationValue != "" {
				return fmt.Errorf("only one client with CNI network annotation value is supported")
		
			}
			cniNetworksAnnotationValue = annotationValue
		}
	}
```